src/components/forms/Combobox.tsx

```diff
@@ -758,22 +760,28 @@ export function ComboboxAsync<T = any>({
   const [entries, setEntries] = useState<Entry<T>[]>([]);
   const [url, setUrl] = useState(endpoint);
 
-  const { data } = useQuery(
-    [new URL(url).pathname, new URL(url).href],
-    () => {
-      const $url = new URL(url);
+  const enableQueryTimeOut = useRef<NodeJS.Timeout | undefined>(undefined);
 
-      if (sortBy) {
-        $url.searchParams.set('sort', sortBy);
-      }
+  const [enableQuery, setEnableQuery] = useState<boolean>(false);
```

Comment:
Can you walk me through this, why do we need timeouts?

Reply:
@beganovich Let's provide you with an example. For instance, navigate to the Task creation page.

- Select the project.
- Visit the Projects page and remove the project that has already been selected on the Task creation page.
- Return to the Task creation page. If you debug the Combobox behavior, you'll notice that the initial query is executed using the `project_id` selected in the first step. However, after some re-rendering and updating of the task's state, you'll find that the `project_id` is an empty string. This change occurs after the re-render, but the query has already been executed.

Apart from the issue of double querying, you may experience unusual behavior on the screen. It may appear list of projects once with the deleted project and the second time without it. The reason the project appears the first time in the list is because the old `project_id` value is sent as `with` parameter.

There are two possible solutions: One option is to navigate through all pages, and upon unmounting components, reset the value to its default, thus avoiding any unsaved changes. Alternatively, this can be handled directly within the Combobox component.

In essence, this involves waiting for the final value of `inputOptions.value` during the initial mount of the Combobox component for the specific page.

Let me know your thoughts.

Reply:
@beganovich However, aside from resolving the double querying bug, this also clarifies the behavior of setting the Combobox value. When testing various scenarios, you may notice instances where we experience set, remove, and then set the value in the Combobox again. This occurs due to the double querying and the `inputOptions.value` changing once or twice before obtaining the correct value.